components/BarcodeScanner.vue

```diff
@@ -66,10 +69,13 @@ export default {
       cameraList.value = availableCameras.map(c => {
         return {deviceId: c.id, label: c.label};
       });
-
-      // select last camera in list to attempt to default to a back-facing,
-      // non-fish-eye camera
-      const defaultCameraId = cameraList.value.at(-1).deviceId;
+      // Attempt to default to a back-facing, non-fish-eye camera
+      // Most often this is the last one in the list of back cameras
+      const backCameras = cameraList.value.filter(camera => {
+        return camera.label.toLowerCase().includes('back');
+      });
+      const defaultCameraId =
+        backCameras?.at(-1)?.deviceId || cameraList.value?.at(-1)?.deviceId;
```

Comment:
Maybe `deviceId` can be `0` (I don't know), `backCameras` will always be an an array (non-falsey), and if it's possible for `cameraList.value` to be `undefined` then we'd need a `?` above too ... but I don't think it can be?

```suggestion
        backCameras.at(-1)?.deviceId ?? cameraList.value.at(-1)?.deviceId;
```

Reply:
All of the device IDs I've seen are super long strings. But there's no telling!